tsinfer/formats.py

```diff
@@ -1315,6 +1315,32 @@ def subset(self, individuals=None, sites=None, **kwargs):
             subset.record_provenance(command="subset", **kwargs)
         return subset
 
+    def get_sites_time_bound(self):
+        """
+        Returns a numpy array of the lower bound of the time of sites in the SampleData
+        file. Each individual with a nonzero time (from the individuals_time array)
+        gives a lower bound on the age of sites where the individual carries a
+        derived allele.
+
+        :return: A numpy array of the lower bound for each sites time.
+        :rtype: numpy.ndarray(dtype=float64)
+        """
+        samples_individual = self.samples_individual[:]
```

Comment:
I think sample data files have to have individuals, but I wonder if we should put an `assert np.all(samples_individual >= 0)` in there just to be safe?